### ISSUE TITLE
Renaming a file updates it in the editor but doesn't rename it on the disk if its only letter's case

### ISSUE THREAD

**Issue by ydeltastar** [godotengine/godot#94340]
### Tested versions

v4.3.beta.custom_build [97b8ad1af]

### System information

Windows 11

### Issue description

If I change only the cases in the name of file using the editor's filesystem dock, the name updates in the interface but the file isn't renamed on the disk.

There is also a cache issue going on because reloading the editor doesn't show the current filename on the disk as well.

### Steps to reproduce

* Create a new script file "player.gd"
* Use the filesystem dock to rename it to "Player.gd". It updates in the editor but doesn't rename the file in the disk.
* Rename it to "Player2.gd". Now it updates in the disk as well.

### Minimal reproduction project (MRP)

N/A

**Comment by ydeltastar** [comment]
The issue doesn't happen when I place a breakpoint somewhere here like `ReplaceFileW` or `_wrename` while debugging.
https://github.com/godotengine/godot/blob/97b8ad1af0f2b4a216f6f1263bef4fbc69e56c7b/drivers/windows/dir_access_windows.cpp#L286-L298
But it always happens if it runs normally. No clue why since nothing returns errors... Maybe file locking or threading issues? I have a very fast SSD NVMe so it's possible that debug breaking gives it time for the issue to untangle but running normally doesn't.

**Comment by AThousandShips** [comment]
This is because of [case sensitive file systems](https://docs.godotengine.org/en/latest/tutorials/best_practices/project_organization.html#case-sensitivity), or rather lack of them, this is an operating system limitation, don't know how to approach it from the editor side

**Comment by 4X3L82** [comment]
> This is because of [case sensitive file systems](https://docs.godotengine.org/en/latest/tutorials/best_practices/project_organization.html#case-sensitivity), or rather lack of them, this is an operating system limitation, don't know how to approach it from the editor side

Yes, this is more on Windows than Godot.
I just tested this on Linux: I renamed a file with just case changes in the editor, and the changes are reflected on disk.

**Comment by AThousandShips** [comment]
One issue here is that the editor *should* support case specific things, as the exported project *does* support case sensitivity (inside the exported `res://` files are case sensitive), but the limitation is that windows doesn't allow renaming a file changing just the casing

Having the editor detect this and warn or reflect it would be useful, but might not be trivial to do

**Comment by AThousandShips** [comment]
Will check the code above and see if there's something broken with it indeed

**Comment by AThousandShips** [comment]
I can't replicate this myself, for me it fixes the name correctly on disk, unsure what might not be working for you

**Comment by LikiaBili** [comment]
> I can't replicate this myself, for me it fixes the name correctly on disk, unsure what might not be working for you, I'm on Windows 11

So the solution is to either the user updates windows or adding a compactable solution like deleting the original file and then creating a new file automatically to achieve the same result

**Comment by AThousandShips** [comment]
This works fine on my Windows 11, the original report is also Windows 11 so this isn't a matter of Windows 10 Vs 11, can you replicate this on windows 11?

**Comment by ydeltastar** [comment]
> I can't replicate this myself, for me it fixes the name correctly on disk, unsure what might not be working for you, I'm on Windows 11

Yeah, that's the weird thing. I can reproduce 100% of the time on any project but I remember it never being a problem before. It started recently but now it happens in previous versions of Godot too.

**Comment by DylanCheetah** [comment]
I am unable to reproduce this bug on Windows 11. I suspect that it was related to Windows itself. I do recall that for a while renaming a file in the File Explorer had the exact same bug as well. However, it appears to be fixed.
